### PR TITLE
Fix cache refresh loop in production mode clusters (#53)

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Fixes bug #53.

(This PR is likely to cause a merge conflict with PR #49 (in the bucket and bucket-type templates)).

### CODE REVIEW

File: app/pods/bucket-type/route.js [hunk 1]
```diff
@@ -16,7 +16,13 @@ export default Ember.Route.extend({
...
     setupController: function(controller, model) {
         this._super(controller, model);
-
+        if(!model.get('bucketList')) {
+            // Init to empty bucket list
+            let buckets = null;
+            let emptyList = this.explorer.createBucketList(buckets, model.get('cluster'),
+                model.get('bucketType'), this.store);
+            model.set('bucketList', emptyList);
+        }
```
Comment (paultannenbaum) [inline app/pods/bucket-type/route.js]: Does it make sense to have this logic here versus adding it to the `explorer. getBucketTypeWithBucketList` method?

**Comment by paultannenbaum** [comment]
Looks good to me. The only question/concern is with the `setupController`methods on bucket types and bucket routes. Isn't that logic for creating the empty lists already handled in the explorer service calls?

**Comment by dmitrizagidulin** [comment]
Let me see here, if putting the init empty list logic into the service instead of setupController works... Checking.

**Comment by dmitrizagidulin** [comment]
Right, so, the reason it's in setupController, is because of the asynchrony of the `explorer.getBucketTypeWithBucketList` method. In my initial implementation, there was no section in `setupController`, with the reasoning that the empty list will be initialized in `getBucketTypeWithBucketList`. 

The problem is, the first time an empty bucket list is initialized is on line 796 of explorer.js, _after_ the `.getBucketList()` promise resolves. But that resolution can take several seconds (if retrieving a large list), and there's no way to display the initial status message until that happens.

Hence, the setupController logic was added, explicitly.

**Comment by paultannenbaum** [comment]
From my understanding of how Ember interacts with promises, the `setupController` will not be invoked until after the model has fully resolved. So for something like bucket types, this means that `setupController` will not be run until after `getBucketTypeWithBucketList` has fully resolved since it is the last thing returned in the `model` method of the route. So I think this code might be unnecessary, unless I am still misunderstanding.

**Comment by dmitrizagidulin** [comment]
@paultannenbaum Per our conversation on chat, I moved the initialization of the empty lists from `setupController` into the models' `init` methods.

**Comment by paultannenbaum** [comment]
+1 Looks good to me. Nice job!

**Pull request merged**